test/integration/dss/test_dss_api.py

```diff
@@ -21,6 +21,18 @@
 from test import reset_tweak_changes, TEST_DIR
 
 
+def get_uploaded_file_names(path: str):
+    files = []
+    for entry in os.scandir(path=path):
+        if entry.is_file():
+            files.append(entry.name)
+        if entry.is_dir():
+            nested_files = get_uploaded_file_names(entry.path)
```

Comment:
This is building an excessive number of temporary strings and lists during recursion. Think about how you can make this more efficient. Also check if there isn't built-in functionality for performing a recursive listing of files.